src/operator/nn/mkldnn/mkldnn_base-inl.h

```diff
@@ -153,9 +153,8 @@ static inline bool SupportMKLDNN(int dtype, const mxnet::TShape &shape) {
     // MKLDNN currently does not support 0-dim Tensor and 0-size Tensor
     return false;
   }
-
   return (dtype == mshadow::kFloat32 || dtype == mshadow::kBfloat16) &&
-         (ndim == 1 || ndim == 2 || ndim == 4);
```

Comment:
What's the issue with 3D tensor?

Reply:
No. This is not needed for this PR. We have already enabled mkldnn conv with 3D tensor previously.

Reply:
can you point me to where it's handled? I didn't understand the separate treatment of 3D